FinancialTransactionsApi/V1/Infrastructure/Entities/TransactionDbEntity.cs

```diff
@@ -9,6 +9,9 @@ namespace FinancialTransactionsApi.V1.Infrastructure.Entities
     public class TransactionDbEntity
     {
         [DynamoDBHashKey]
+        [DynamoDBProperty(AttributeName = "pk")]
```

Comment:
Why is "pk" introduced, instead of having ID as a hash key?

Reply:
Dynamo queries require at least the hash key of the table you want to query. I created a composite key(combination of pk which is constant as partition key and id which is dynamic as sort key). Since we are no longer using GSI the best way to achieve close to what we want is the use QueryAsync that requires a key